clang/lib/Driver/ToolChains/Darwin.cpp

```diff
@@ -2494,6 +2494,19 @@ void DarwinClang::AddClangCXXStdlibIncludeArgs(
                    << "\"\n";
     }
 
+    // Check for the folder where the executable is located, if different.
+    if (getDriver().getInstalledDir() != getDriver().Dir) {
+      InstallBin = llvm::StringRef(getDriver().Dir.c_str());
+      llvm::sys::path::append(InstallBin, "..", "include", "c++", "v1");
+      if (getVFS().exists(InstallBin)) {
+        addSystemInclude(DriverArgs, CC1Args, InstallBin);
+        return;
+      } else if (DriverArgs.hasArg(options::OPT_v)) {
+        llvm::errs() << "ignoring nonexistent directory \"" << InstallBin
+                     << "\"\n";
+      }
+    }
+
     // Otherwise, check for (2)
     llvm::SmallString<128> SysrootUsr = Sysroot;
     llvm::sys::path::append(SysrootUsr, "usr", "include", "c++", "v1");
```

Comment:
Not attached to this line: please add tests for this new behavior.

Reply:
Thank you, Louis @ldionne.

If I understand your review right, the comments need a small update. That shouldn't be a problem.

As for the tests, I have no experience with this testing infrastructure, but I'll try to figure out a solution.

Reply:
I think you probably want to add new tests in `clang/test/Driver/darwin-header-search-libcxx.cpp`. You also mentioned that other platforms (e.g. Linux?) handle search paths this way, right? Can you show where this is handled? They might have some tests you could take inspiration from?

Reply:
Test added.